File: isomorphic.js
// Time Complexity : O(N) length of the string
// Space Complexity : O(1)
// Did this code successfully run on Leetcode :Yes
// Any problem you faced while coding this : No

//Use two hashmaps to hold the key value pairs
// compare if key already exists, else append
//check if the key value are the same in both hashmaps, return true. 


/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isIsomorphic = function(s, t) {
    if(s.length != t.length){
        return false
    }
    let sMap = new Map(); 
    let tMap = new Map();
    for(let i = 0; i < s.length; i++){
        let sChar = s.charAt(i);
        let tChar = t.charAt(i);
        if(sMap.has(sChar)){
            if(sMap.get(sChar) != tChar) return false;
        }
        else {
            sMap.set(sChar, tChar)
        }
        
         if(tMap.has(tChar)){
            if(tMap.get(tChar) != sChar) return false;
        }
        else {
            tMap.set(tChar, sChar)
        }
    }
    return true;
    
};

File: Isomorphic.cs
// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : yes
// Any problem you faced while coding this : no

public class Solution {
    public bool IsIsomorphic(string s, string t) {        
        if(s.Length != t.Length){ return false;}

        Dictionary<Char, Char> sMap = new Dictionary<Char, Char>();
        Dictionary<Char, Char> tMap = new Dictionary<Char, Char>();

        for(int i=0; i < s.Length; i++)
        {
            Char sChar = s[i];
            Char tChar = t[i];

            if(!sMap.ContainsKey(sChar))
            {
                sMap[sChar] = tChar;
            }else
            {
                if(sMap[sChar] != tChar)
                    return false;
            }

            if(!tMap.ContainsKey(tChar))
            {
                tMap[tChar] = sChar;
            }else
            {
                if(tMap[tChar] != sChar)
                    return false;
            }
        }

        return true;

    }
}
